### ISSUE TITLE
Json dump of a liquid hash is invalid.

### ISSUE THREAD

**Issue by syte** [mikebridge/Liquid.NET#67]
Currently whenever I dump a liquid hash,  it serializes empty enumerables into nothing. Also, enumerables aren't wrapped in brackets, and each item in the list isn't being separated from the next with a comma.

**Example:**
`{
     facets: ,
     results: {
         name: 'Test'
      },
      {
        name: 'Test2'
     }
}`

**Comment by mikebridge** [comment]
Is that from just putting a hash directly into a template?  I don't think I really covered that case---I have a  [default stringification] (https://github.com/mikebridge/Liquid.NET/blob/master/Liquid.NET/src/Constants/LiquidHash.cs#L130) method for that but it is not very robust.

I can probably add a hook for a JSON serializer into the initialization so that it doesn't take a dependency on NewtonSoft.

**Comment by syte** [comment]
@mikebridge , yes it's from putting the hash directly into the template. I do like your idea of being able to add a hook for serialization.
